File: src/com/leozhi/topic28/Solution03.java
package com.leozhi.topic28;

/**
 * @author leozhi
 * 比较字符串
 * 通过
 * 0ms
 */
public class Solution03 {
    public int strStr(String haystack, String needle) {
        int len1 = haystack.length(), len2 = needle.length();
        for (int start = 0; start < len1 - len2 + 1; start++) {
            if (haystack.substring(start, start + len2).equals(needle)) {
                return start;
            }
        }
        return -1;
    }
}

File: src/com/leozhi/topic455/Solution03.kt
package com.leozhi.topic455

/**
 * @author leozhi
 * 贪心算法
 * 通过
 * 284ms
 */
class Solution03 {
    fun findContentChildren(g: IntArray, s: IntArray): Int {
        val len1 = g.size
        val len2 = s.size
        var child = 0
        var cookie = 0
        g.sort()
        s.sort()
        while (child < len1 && cookie < len2) {
            if (g[child] <= s[cookie]) {
                ++child
            }
            ++cookie
        }
        return child
    }
}